neo4j-jdbc-bolt/src/main/java/org/neo4j/jdbc/bolt/BoltPreparedStatement.java

```diff
@@ -118,7 +118,7 @@ private StatementResult executeInternal() throws SQLException {
 	}
 
 	private boolean hasResultSet() {
-		return this.statement != null && this.statement.toLowerCase().contains("return");
+		return this.statement != null && this.statement.toLowerCase().contains(" return ");
```

Comment:
there is a query-type in result that we should check against. http://neo4j.com/docs/api/java-driver/current/org/neo4j/driver/v1/summary/ResultSummary.html#statementType--